analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1549 @@
+# write_parquet(shap_comparison, "shap_comparison.parquet")
+# write_parquet(metadata_comparison, "metadata_comparison.parquet")
+```
+
+```{r}
+assessment_data <- read_parquet("assessment_data.parquet")
+assessment_pin <- read_parquet("assessment_pin.parquet")
+assessment_card <- read_parquet("assessment_card.parquet")
+performance <- read_parquet("performance.parquet")
+shap <- read_parquet("shap.parquet")
+metadata <- read_parquet("metadata.parquet")
+assessment_pin_comparison <- read_parquet("assessment_pin_comparison.parquet")
+model_performance_assessment_comparison <- read_parquet("model_performance_assessment_comparison.parquet")
+shap_comparison <- read_parquet("shap_comparison.parquet")
+metadata_comparison <- read_parquet("metadata_comparison.parquet")
+```
+
+
+
+```{r data_processing}
+assessment_data_small <- assessment_data %>%
+  select(meta_pin, meta_card_num, meta_nbhd_code, loc_longitude, loc_latitude, meta_township_name, !!sym(params$added_feature))
+
+# Create a card level dataset
+
+working_data_card <- shap %>%
+  select(meta_pin, meta_card_num, pred_card_shap_baseline_fmv, !!sym(params$added_feature)) %>%
+  rename(!!params$added_feature_shap := !!sym(params$added_feature)) %>%
+  inner_join(assessment_card, by = c("meta_pin", "meta_card_num")) %>%
+  rename(added_feature_card = !!sym(params$added_feature)) %>%
+  inner_join(assessment_data_small, by = c("meta_pin", "meta_card_num")) %>%
+  group_by(meta_nbhd_code) %>%
+  mutate(
+    !!paste0(params$added_feature, "_shap_neighborhood_mean") := mean(abs(!!sym(params$added_feature_shap)), na.rm = TRUE),
+    !!paste0(params$added_feature, "_shap_neighborhood_90th") := quantile(abs(!!sym(params$added_feature_shap)), 0.9, na.rm = TRUE),
+    median_card_value = median(pred_card_initial_fmv, na.rm = TRUE)
+  ) %>%
+  ungroup() %>%
+  inner_join(assessment_pin %>% select(meta_pin, pred_pin_final_fmv, pred_pin_initial_fmv), by = "meta_pin") %>%
+  rename(
+    pred_pin_final_fmv_new = pred_pin_final_fmv,
+    pred_pin_initial_fmv_new = pred_pin_initial_fmv
+  ) %>%
+  inner_join(assessment_pin_comparison %>% select(meta_pin, pred_pin_final_fmv, pred_pin_initial_fmv), by = "meta_pin") %>%
+  rename(
+    pred_pin_final_fmv_comp = pred_pin_final_fmv,
+    pred_pin_initial_fmv_comp = pred_pin_initial_fmv
+  ) %>%
+  mutate(
+    shap_relative = percent((!!sym(params$added_feature_shap) / pred_card_initial_fmv), accuracy = 0.01),
+    diff_pred_pin_final_fmv = round(pred_pin_final_fmv_new - pred_pin_final_fmv_comp, 2),
+    pred_pin_final_fmv_new = scales::dollar(pred_pin_final_fmv_new),
+    pred_pin_final_fmv_comparison = scales::dollar(pred_pin_final_fmv_comp),
+    diff_pred_pin_initial_fmv = round(pred_pin_initial_fmv_new - pred_pin_initial_fmv_comp, 2),
+    pred_pin_initial_fmv_new = scales::dollar(pred_pin_initial_fmv_new),
+    pred_pin_initial_fmv_comp = scales::dollar(pred_pin_initial_fmv_comp)
+  )
+
+temp_pin <- working_data_card %>%
```

Comment:
Creates the pin level characteristics that we need (which need neighborhood grouping)